containers/buildah
# Add extra debug info and JSON logging utility

#### What type of PR is this?



> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 

/kind feature

> /kind flake
> /kind other

#### What this PR does / why we need it:

This adds extra debug logs for the images cache info state and events, like found/updated an entry.
Writes details into debug logs as "safe" JSON that may also be post-processed by external pretty view tooling,
like ``jq``.
    
In order to log internal cache structures/entries as JSON, it creates an extra LogJSON utility.

#### How to verify it

To see results with added logs, use ``sudo buildah --log-level=debug bud --loglevel=3 ...``

#### Which issue(s) this PR fixes:





Related #2782 

#### Special notes for your reviewer:

In order to make structures logged as JSON, its data fields need to be exported for JSON Marshalling to work.
This patch provides re-esporting func for a particular structure to debug it with the new logger.
Alternative approach is to change the original structure to have its data fields exported. But that would have introduced much
more impact to other places that rely on it...

#### Does this PR introduce a user-facing change?



```release-note
In the builah bud debug mode, extra information on images caching is available in the produced logs
```
